ovn_cluster.sh

```diff
@@ -195,75 +203,90 @@ function add-ovs-container-ports() {
-            ./ovs-runc add-port $br $eth ${CENTRAL_NAME}-2 --ipaddress=${ip2}/${cidr}
-            echo $ip2 > _ovn_central_2
+        for i in $(seq 3); do
+            echo -n > _ovn_central_$i
+        done
+        echo -n > _ovn_remote
 
-            (( ip_index += 1))
-            ip3=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
-            ./ovs-runc add-port $br $eth ${CENTRAL_NAME}-3 --ipaddress=${ip3}/${cidr}
-            echo $ip3 > _ovn_central_3
-            echo "${REMOTE_PROT}:$ip1:6642,${REMOTE_PROT}:$ip2:6642,${REMOTE_PROT}:$ip3:6642" > _ovn_remote
-        else
-            ./ovs-runc add-port $br $eth ${CENTRAL_NAME} --ipaddress=${ip}/${cidr}
-            echo "${REMOTE_PROT}:$ip:6642" > _ovn_remote
-        fi
+        for name in "${CENTRAL_NAMES[@]}"; do
+            if [ "$OVN_DB_CLUSTER" = "yes" ]; then
+                ip1=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
+                ./ovs-runc add-port $br $eth ${name}-1 --ipaddress=${ip1}/${cidr}
+                echo "$name $ip1" >> _ovn_central_1
+                (( ip_index += 1))
+
+                ip2=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
+                ./ovs-runc add-port $br $eth ${name}-2 --ipaddress=${ip2}/${cidr}
+                echo "$name $ip2" >> _ovn_central_2
+                (( ip_index += 1))
+
+                ip3=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
+                ./ovs-runc add-port $br $eth ${name}-3 --ipaddress=${ip3}/${cidr}
+                echo "$name $ip3" >> _ovn_central_3
+                (( ip_index += 1))
+
+                echo "$name ${REMOTE_PROT}:$ip1:6642,${REMOTE_PROT}:$ip2:6642,${REMOTE_PROT}:$ip3:6642" >> _ovn_remote
+            else
+                ip=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
+                ./ovs-runc add-port $br $eth ${name} --ipaddress=${ip}/${cidr}
+                echo "$name ${REMOTE_PROT}:$ip:6642" >> _ovn_remote
+                (( ip_index += 1))
+            fi
+        done
 
         for name in "${GW_NAMES[@]}"; do
-            (( ip_index += 1))
             ip=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
             ./ovs-runc add-port $br $eth ${name} --ipaddress=${ip}/${cidr}
+            (( ip_index += 1))
         done
 
         if [ "$RELAY_COUNT" -gt 0 ]; then
             relay_remotes=""
             for name in "${RELAY_NAMES[@]}"; do
-                (( ip_index += 1))
                 ip=$(./ip_gen.py $ip_range/$cidr $ip_start $ip_index)
                 ./ovs-runc add-port $br $eth ${name} --ipaddress=${ip}/${cidr}
                 relay_remotes=$relay_remotes",${REMOTE_PROT}:$ip:6642"
+                (( ip_index += 1))
             done
-            orig_remotes=$(cat _ovn_remote)
-            echo "${relay_remotes}" | cut -c 2- > _ovn_remote
+            echo "${relay_remotes}" | cut -c 2- >> _ovn_remote
+            orig_remotes=$(tail -n 1 _ovn_remote)
```

Comment:
This breaks relays because it essentially causes a relay to try to sync from itself.  I'm not sure why these 2 lines were swapped but I think they shouldn't have been.

Reply:
@dceara ack, I will fix it